python/lsst/jointcal/jointcal.py

```diff
@@ -794,6 +796,8 @@ def _iterate_fit(self, associations, fitter, max_steps, name, whatToFit,
             The star/reference star associations to fit.
         fitter : `lsst.jointcal.FitterBase`
             The fitter to use for minimization.
+        model : `lsst.jointcal.PhotometryModel` or `lsst.jointcal.AstrometryModel`
+            The model being fit by ``fitter``.
```

Comment:
I'm not sure about this. First, do you really want to run a (presumably expensive) non-negativity check on each iteration? Second, do you really need to change the `_iterate_fit` interface to do it? Surely the fitter knows what model it's fitting, and you can just ask it.

Reply:
The check isn't very expensive: it only tests a dozen or so points per ccdImage. I do want to run it every iteration, so that we can see immediately when something goes wrong.

Fair point on the latter: it's not as pretty as I'd like because there's not a parent class for `AstrometryModel` and `PhotometryModel`, but we do plan to do that eventually. I've cleaned this up.